x/gov/client/rest/rest.go

```diff
@@ -77,6 +77,11 @@ type voteReq struct {
 
 func postProposalHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if !cliCtx.AllowUnsafe {
```

Comment:
I initially went with a `ClientRouter` wrapper struct that contained a `mux` router and an unsafe `bool`, but found the route registration ugly as there was a giant DUP if/else switch.

The following solution leaves it up to the actual handler and relies on the context. Also, not super clean, but I feel like it is more DRY compared to the previous solution.